tmpl/blog/introducing-charrua-dhcp.md

```diff
@@ -0,0 +1,221 @@
+should be available soon, with irmin and other backends. The main idea is to
+always return a new lease database at every input, or maybe just the updates to
+be applied, in this scenario, the caller would be able to store the database in
+permanent storage as he sees fit.
+
+#### Configuration
+
+This project started independently of MirageOS, and at that time, the best
+configuration I could think of was the well known `ISC` `dhcpd.conf`, therefore
+the configuration uses the same format, but it does not support all the myriad
+of options the original one does.
+
+```ocaml
+  type t = {
+    addresses : (Ipaddr.V4.t * Macaddr.t) list;
+    subnets : subnet list;
+    options : Dhcp_wire.dhcp_option list;
+    hostname : string;
+    default_lease_time : int32;
+    max_lease_time : int32;
+  }
+
+  val parse : string -> (Ipaddr.V4.Prefix.addr * Macaddr.t) list -> t
+  (** [parse cf l] Creates a server configuration by parsing [cf] as an ISC
+      dhcpd.conf file, currently only the options at [sample/dhcpd.conf] are
+      supported. [l] is a list of network addresses, each pair is the output
+      address to be used for building replies and each must match a [network
+      section] of [cf]. A normal usage would be a list of all interfaces
+      configured in the system *)
+```
+
+Although it is a great format, it doesn't play exactly nice with MirageOS and
+OCaml, since the unikernel needs to parse a string at runtime to build the
+configuration, this requires a file IO backend and other complications. The
+next version should provide OCaml helpers for building the configuration, which
+would drop the requirements of a file IO backend and facilitate writing tests.
+
+### Building a simple server
+
+The easiest way is to follow [mirage-skeleton DHCP
+README](https://github.com/mirage/mirage-skeleton/blob/master/dhcp/README.md).
+
+### Future
+
+The next steps would be:
+
+* Provide helpers for building the configuration.
+* Expose the lease database in an immutable structure, possibly a `Map`, adding
+also support/example for [Irmin](https://github.com/mirage/irmin).
+* Use [Functoria] (https://github.com/mirage/functoria) to pass down the
+configuration in [mirage-skeleton]
+(https://github.com/mirage/mirage-skeleton/blob/master/dhcp/README.md). Currently
+it is awkward since the user has to edit `unikernel.ml` and `config.ml`, with
+[Functoria] (https://github.com/mirage/functoria) we would be able to have it
+much nicer and only touch `config.ml`.
+* Convert MirageOS DHCP client code to use [Dhcp_wire]
+(http://haesbaert.github.io/charrua-core/api/Dhcp_wire.html), or perhaps add a
+client logic functionality to [Charrua]
+(http://www.github.com/haesbaert/charrua-core).
+
+### Finishing words
+
+This is my first real project in OCaml and I'm more or less a newcomer to
+functional programming as well, my background is mostly kernel hacking as an
+ex-openbsd developer.
```

Comment:
> Btw, I'd love to hear if people are actually using it and what are the problems they're finding

I'm about to merge the post so I'll add some words explicitly asking for user feedback and will link to the [issue tracker](https://github.com/haesbaert/charrua-core/issues).